src/others/observable/lv_observable.h

```diff
@@ -0,0 +1,263 @@
+
+/**
+ * Remove all observers of a specified widget from the list of observers
+ * @param subject address of the subject
+ * @param obj the LVGL widget
+ */
+void lv_subject_unsubscribe_obj(lv_subject_t* subject, lv_obj_t* obj);
+
+/**
+ * Remove all observers of a subject
+ * @param subject address of the subject
+ */
+void lv_subject_unsubscribe_all(lv_subject_t* subject);
+
+/**
+ * Update the value of an int type subject
+ * @param subject address of the subject
+ * @param value new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_int(lv_subject_t* subject, int32_t value);
+
+/**
+ * Update the value of a pointer type subject
+ * @param subject address of the subject
+ * @param value new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_ptr(lv_subject_t* subject, void* value);
+
+/**
+ * Update the value of a color type subject
+ * @param subject address of the subject
+ * @param color new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_color(lv_subject_t* subject, lv_color_t color);
+
+/**
+ * Copy the string into the buffer of a string type subject
+ * @param subject address of the subject
+ * @param buf string to copy 
+ * @note automatically notifies all observers
+ */
+void lv_subject_copy_string(lv_subject_t* subject, char* buf);
+
+/**
+ * Set bitmask and value comparison pattern of the observer
+ * @param observer address of the observer
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ * @note The value
+ */
+void lv_observer_bitmask_set_cond(lv_observer_t* observer, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object flag
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param flag the affected flag
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_flag(lv_subject_t* subject, lv_obj_t* obj, lv_obj_flag_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object state
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param state the affected state
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_state(lv_subject_t* subject, lv_obj_t* obj, lv_state_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind int value to a local style property
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param prop the style property
+ * @param selector the part/state selector
+ */
+void* lv_bind_int_to_obj_local_style_prop(lv_subject_t* subject, lv_obj_t* obj, lv_style_prop_t prop, lv_style_selector_t selector);
```

Comment:
I find it too special to have only `int`s. I would remove it in the first pass, a reconsider it in a new PR.

Reply:
We could rename it to 'lv_bind_subject_to_obj_local_style_prop', since the subject value can contain other types as well, and there is no 'int' specific in this function.

Reply:
How can we decide which value to pass to the style?

Reply:
I believe we can simply cast the lv_subject_value_t to lv_style_value_t.

Reply:
`float` needs to removed anyway, because LVGL is integer only. Without it the  `lv_subject_value_t` and `lv_style_value_t` will be quite the same.
 
In v9 
- there will be a `LV_USE_FLOAT` option to enable then on some places. 
- we can use a single `lv_value_t` on both places. 

However in v8 we should align this feature to what we already have there.